File: MediaBase/MediaBase/MovieInfo.m
//
//  MovieInfo.m
//  MediaBase
//
//  Created by tbago on 16/12/16.
//  Copyright © 2016年 tbago. All rights reserved.
//

#import "MovieInfo.h"

@interface MovieInfo()
@property (strong, nonatomic) NSMutableArray        *innerStreamArray;
@property (strong, nonatomic) NSMutableDictionary   *innerMetaData;
@end

@implementation MovieInfo

- (instancetype)init {
    self = [super init];
    if (self) {
        
    }
    return self;
}

#pragma mark - public method

- (void)addStreamInfoToMovieInfo:(StreamInfo *)streamInfo {
    [self.innerStreamArray addObject:streamInfo];
}

- (void)addMetaDataToMovieInfo:(NSString *)key
                         value:(NSString *)value {
    self.innerMetaData[key] = value;
}

#pragma mark - get & set

- (NSArray *)streamArray {
    return _innerStreamArray;
}

- (NSDictionary *)metaData {
    return _innerMetaData;
}

- (NSMutableArray *)innerStreamArray {
    if (_innerStreamArray == nil) {
        _innerStreamArray = [[NSMutableArray alloc] init];
    }
    return _innerStreamArray;
}

- (NSMutableDictionary *)innerMetaData {
    if (_innerMetaData == nil) {
        _innerMetaData = [[NSMutableDictionary alloc] init];
    }
    return _innerMetaData;
}
@end

File: MediaBase/MediaBase/MovieInfo.cc
//
//  MovieInfo.cc
//  MediaBase
//
//  Created by tbago on 16/12/16.
//  Copyright © 2021 tbago. All rights reserved.
//

#include "MovieInfo.h"

namespace media_base {

MovieInfo::MovieInfo()
{
    id = -1;
}

MovieInfo::MovieInfo(const MovieInfo &movieInfo)
{
    CopyInfo(movieInfo);
}

MovieInfo & MovieInfo::operator=(const MovieInfo &movieInfo)
{
    if (this == &movieInfo) {
        return *this;
    }
    CopyInfo(movieInfo);
    return *this;
}

MovieInfo::~MovieInfo()
{
    for (uint32_t i = 0; i < streams.size(); i++) {
        StreamInfo *streamInfo = streams[i];
        delete streamInfo;
        streams[i] = NULL;
    }
    streams.clear();
}

void MovieInfo::CopyInfo(const MovieInfo &movieInfo)
{
    id = movieInfo.id;
    name = movieInfo.name;
    format = movieInfo.format;

    for (uint32_t i = 0; i < movieInfo.streams.size(); i++) {
        StreamInfo *streamInfo = new StreamInfo(*movieInfo.streams[i]);
        streams.push_back(streamInfo);
    }
}

}   // namespoace media_base

